text/0000-feature-metadata.md

```diff
@@ -0,0 +1,275 @@
+-   Added Cargo arguments if escape hatches for `public` or `unstable` are created
+-   Docstrings can be lengthy, adding noise to `Cargo.toml`. This could
+    potentially be solved with the below mentioned `doc-file` key.
+-   `unstable` and `public` uses may not be common enough to be worth including
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+-   TOML-docstrings (`## Some doc comment`) and attributes that mimic Rust
+    docstrings could be used instead of a `doc` key. This is discussed in
+    [future-possibilities], but decided against to start since it is much simpler
+    to parse standard TOML.
+-   Feature descriptions could be specified somewhere in Rust source files. This
+    has the downside of creating multiple sources of truth on features.
+
+# Prior art
+
+[prior-art]: #prior-art
+
+-   There is an existing crate that uses TOML comments to create a features table:
+    <https://docs.rs/document-features/latest/document_features/>
+-   `docs.rs` displays a feature table, but it is fairly limited
+-   Ivy has a [visibility attribute] for its configuration (mentioned in [cargo #10882])
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+-   Should `foo = {}` be allowed, or is `foo = { requires = [] }` the minimum if
+    only using this field? I am leaning toward having it required to avoid
+    inconsistency like
+
+    ```toml
+    foo = {}
+    bar = []
+    baz = {}
+    ```
+
+    However, I would also say that `foo = { deprecated = true }` (no `requires`
+    key) could be valid.
+
+-   Do we want all the proposed keys? Specifically, `public` and `unstable` may be
+    more than what is needed.
+
+    See also:
+
+    -   <https://github.com/rust-lang/cargo/issues/10882>
+    -   <https://github.com/rust-lang/cargo/issues/10881>
+
+-   If we use the semantics as-written, should there be a
+    `--allow-private-features` or `--allow-unstable-features` flag? Or how should
+    a user opt in?
+-   Does it make sense to have separate `hidden` (not documented) and `public`
+    (feature not allowed downstream) attribute? I think probably not
+-   Should there be a way to deny deprecated features?
+
+It is worth noting that simpler keys (`requires`, `doc`, `deprecated`) could be
+stabilized immediately and other features could be postponed.
+
+# Future possibilities
+
+[future-possibilities]: #future-possibilities
+
+-   Rustdoc will gain the ability to document features. This is planned in an
+    associated RFC.
+-   One option is to allow giving the feature documentation in a separate file,
+    allowing a markdown section specifier. This could be a good option for
+    features requiring long descriptions.
+
+    ```toml
+    foo = { requires = [], doc-file = "features.md#foo" }
+    bar = { requires = [], doc-file = "features.md#bar" }
+    ```
+
+-   Cargo could parse doc comments in `Cargo.toml`, like the above linked
```

Comment:
imo this is unlikely enough that I would not put it in Future Possibilities but Drawbacks (as not supported) or Alternatives.

The reason why I personally feel this is unlikely to ever happen is that you can no longer parse a manifest with a conformant TOML parser nor can you use serde.  You will have to use a format-preserving parser to manually extract every field and some of the comments.

This works in Rust because we own the format and parser.

Reply:
I don't disagree, I have moved this to alternatives